core/src/main/java/com/linecorp/armeria/client/HttpClientBuilder.java

```diff
@@ -38,13 +38,13 @@
     /**
      * An undefined {@link URI} to create {@link HttpClient} without specifying {@link URI}.
      */
-    private static final URI UNDEFINED_URI = URI.create("none+http://undefined");
+    private static final URI UNDEFINED_URI = URI.create("http://undefined");
 
     /**
      * Returns {@code true} if the specified {@code uri} is an undefined {@link URI}.
      */
     static boolean isUndefinedUri(URI uri) {
-        return UNDEFINED_URI == uri;
+        return UNDEFINED_URI.equals(uri);
```

Comment:
Revert this. The reference comparison is intended. 😅

Reply:
If I revert this, then isUndefinedUri(uri) will always return false, as == is comparing if they are exactly the same instance.
Due to this change (normalizeURI method), we will create __a new instance of URI__ by removing none+ from the URI. 
Also, I stumbled on this - because 2 test cases were failing. One of them was HttpClientRequestPathTest.default_withRelativePath. And one more from same test class (sorry don't remember the method name)

Reply:
Thanks for the detail explanations. 🙏 I thought you encountered a problem and fixed this.

How about skipping URI normalization on `UNDEFINDED`? Because it is used only internally. 😀

Reply:
@ikhoon I think that might be an additional complexity - additional branch - for an edge case (I mean, how often will some one create a client and not specify the hostname/authority).

If you are certain, we would have to do - I am more than happy to make the change.

Reply:
> I think that might be an additional complexity - additional branch - for an edge case 

I understand what you are worried about. But `UNDEFINED_URI` is created to build `HttpClient` without hostname. #1143 #1961

I'm still sure to revert this. :-)

Reply:
Cool! Thank you for the links.
I have made required changes, to handle special case around__UNDEFINED_URI__